### PR TITLE
LPS-109882 add test

### PR DESCRIPTION
**Pull request by leoadb**

### CODE REVIEW

**Comment by leoadb** [comment]
ci:test:sf

**Comment by leoadb** [comment]
ci:test:app-builder

**Comment by leoadb** [comment]
cc @jeyvison @brunofarache

File: modules/apps/data-engine/data-engine-taglib/src/main/java/com/liferay/data/engine/taglib/servlet/taglib/util/DataLayoutTaglibUtil.java [hunk 1]
```diff
@@ -303,15 +293,8 @@ private DataLayout _getDataLayout(
...
-			_portal.getHost(httpServletRequest),
-			httpServletRequest.getServerPort(), httpServletRequest.getScheme()
-		).header(
-			"Cookie",
-			"JSESSIONID=" +
-				CookieKeys.getCookie(httpServletRequest, CookieKeys.JSESSIONID)
-		).parameter(
-			"p_auth", AuthTokenUtil.getToken(httpServletRequest)
+		).user(
+			_portal.getUser(httpServletRequest)
```
Comment (leoadb) [inline modules/apps/data-engine/data-engine-taglib/src/main/java/com/liferay/data/engine/taglib/servlet/taglib/util/DataLayoutTaglibUtil.java]: @ealonso that is how you need to call the resource. Use the resource from rest-api not from client.

**Comment by ealonso** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by leoadb** [comment]
ci:test:app-builder

**Comment by leoadb** [comment]
ci:stop

**Comment by leoadb** [comment]
ci:test:app-builder

**Comment by ealonso** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by leoadb** [comment]
ci:test:app-builder

**Comment by leoadb** [comment]
ci:test:app-builder

**Comment by leoadb** [comment]
ci:test:app-builder

**Comment by leoadb** [comment]
ci:test:app-builder

File: modules/apps/data-engine/data-engine-rest-test/src/testIntegration/java/com/liferay/data/engine/rest/resource/v2_0/factory/test/DataDefinitionResourceFactoryImplTest.java [hunk 2]
```diff
@@ -127,7 +156,190 @@ public void testHttpServletRequest() throws Exception {
 		dataDefinitionResource.getDataDefinitionDataDefinitionFieldFieldTypes();
 	}
 
-	@Inject(type = JSONFactory.class)
+	@Test
+	public void testPostDataDefinition() throws Exception {
```
Comment (leoadb) [inline modules/apps/data-engine/data-engine-rest-test/src/testIntegration/java/com/liferay/data/engine/rest/resource/v2_0/factory/test/DataDefinitionResourceFactoryImplTest.java]: @ealonso please see this test. You don't need the DDM api part.

**Comment by leoadb** [comment]
@jeyvison real errors?

**Comment by brunofarache** [comment]
@leoadb almost all tests in app builder are green lately. it seems pretty broken in this PR, have you tested it manually?

**Comment by leoadb** [comment]
@brunofarache I've found the issue and fixed it. Will send a new PR. Thanks!

**Pull request closed without merging**